commands/assign-dish.js

```diff
@@ -0,0 +1,35 @@
+const { SlashCommandBuilder } = require('@discordjs/builders');
+const { getMenu, saveMenu } = require('./../menu-memory/menu.js');
+
+module.exports = {
+	data: new SlashCommandBuilder()
+		.setName('assign-dish')
+		.setDescription('assign a dish to a user')
+		.addUserOption(option => option.setName('user').setDescription('Who is bringing the dish>'))
+		.addStringOption(option => option.setName('dish').setDescription('Enter a dish')),
+	async execute(interaction) {
+		let user = interaction.options.getUser('user');
+		const dish = interaction.options.getString('dish');
+		let menu = await getMenu();
+		
+		if (dish == undefined) {
+			return interaction.reply(`No dish was provided to sign up for`);
```

Comment:
I like when methods have a single return at the end. Is that not a js best practice or is that your convention?

Reply:
I like to return early, can simplify the logic later on and it technically runs faster. But feel free to refactor that as you are correct in saying that typically a good idea.

Reply:
> It might be rendering weird on my phone, how many spaces per tab is in these files?

I use actual tabs to match the documentation and examples online.  And GitHub normally defaults to 8

Reply:
Nah I was just curious. Can always refractor later if its bloated

Reply:
> > It might be rendering weird on my phone, how many spaces per tab is in these files?
> 
> I use actual tabs to match the documentation and examples online. And GitHub normally defaults to 8

You should not use actual tabs ever. Tab button = space means we all see the same thing

Reply:
> > > It might be rendering weird on my phone, how many spaces per tab is in these files?
> > 
> > 
> > I use actual tabs to match the documentation and examples online. And GitHub normally defaults to 8
> 
> You should not use actual tabs ever. Tab button = space means we all see the same thing

Matches the documentation for the tooling used to make it work. I’m keeping it this way

Reply:
Well Hodson is a stickler for following Javascript standards that everyone follows. So not only do we need to remove all tabs and use 2 spaces for all indents, but there should only be one return statement, with the exception of input/state validation, which should only occur at the beginning of a function.